flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/Table.java

```diff
@@ -879,4 +879,118 @@
 	 * @return An OverWindowedTable to specify the aggregations.
 	 */
 	OverWindowedTable window(OverWindow... overWindows);
+
+	/**
+	 * Performs a field add operation. Similar to an SQL SELECT statement. The field expressions
+	 * can contain complex expressions, but can not contain aggregations. It will throw an exception
+	 * if the added fields already exist.
+	 *
+	 * <p>Example:
+	 * <pre>
+	 * {@code
+	 *   tab.addColumns("a + 1 as a1, concat(b, 'sunny') as b1")
+	 * }
+	 * </pre>
+	 */
+	Table addColumns(String fields);
+
+	/**
+	 * Performs a field add operation. Similar to an SQL SELECT statement. The field expressions
+	 * can contain complex expressions, but can not contain aggregations. It will throw an exception
+	 * if the added fields already exist.
+	 *
+	 * <p>Scala Example:
+	 *
+	 * <pre>
+	 * {@code
+	 *   tab.addColumns('a + 1 as 'a1, concat('b, "sunny") as 'b1)
+	 * }
+	 * </pre>
+	 */
+	Table addColumns(Expression... fields);
+
+	/**
+	 * Performs a field add operation. Similar to an SQL SELECT statement. The field expressions
+	 * can contain complex expressions, but can not contain aggregations. Existing fields will be
+	 * replaced if {@code replaceIfExist} is true. If the added fields have duplicate field name,
+	 * then the last one is valid.
+	 *
+	 * <p>Example:
+	 * <pre>
+	 * {@code
+	 *   tab.addColumns(true, "a + 1 as a1, concat(b, 'sunny') as b1")
+	 * }
+	 * </pre>
+	 */
+	Table addColumns(boolean replaceIfExist, String fields);
+
+	/**
+	 * Performs a field add operation. Similar to an SQL SELECT statement. The field expressions
+	 * can contain complex expressions, but can not contain aggregations. Existing fields will be
+	 * replaced if {@code replaceIfExist} is true. If the added fields have duplicate field name,
+	 * then the last one is valid.
+	 *
+	 * <p>Scala Example:
+	 * <pre>
+	 * {@code
+	 *   tab.addColumns(true, 'a + 1 as 'a1, concat('b, "sunny") as 'b1)
+	 * }
+	 * </pre>
+	 */
+	Table addColumns(boolean replaceIfExist, Expression... fields);
+
+	/**
+	 * Performs a field rename operation. Similar to an field alias statement. The field expressions
+	 * should be alias expressions, and only the existing fields can be renamed.
+	 *
+	 * <p>Example:
+	 *
+	 * <pre>
+	 * {@code
+	 *   tab.renameColumns("a as a1, b as b1")
+	 * }
+	 * </pre>
+	 */
+	Table renameColumns(String fields);
+
+	/**
+	 * Performs a field rename operation. Similar to an field alias statement. The field expressions
+	 * should be alias expressions, and only the existing fields can be renamed.
+	 *
+	 * <p>Scala Example:
+	 *
+	 * <pre>
+	 * {@code
+	 *   tab.renameColumns('a as 'a1, 'b as 'b1)
+	 * }
+	 * </pre>
+	 */
+	Table renameColumns(Expression... fields);
```

Comment:
I wonder if we need the `renameColumns`? Isn't `addOrReplace` sufficient?

Reply:
Their semantics are different. Rename just changes the name and does not change the way the column is calculated.  `renameColumns('a as 'a1) != addColumns('a as 'a1)`.  because `addColumns('a as 'a1)` will add new column if 'a1 not exist.